src/main/scala/dpla/ingestion3/executors/MappingExecutor.scala

```diff
@@ -146,48 +159,48 @@ trait MappingExecutor extends Serializable {
                        shortName: String,
                        logsBasePath: String,
                        startTime: Long,
-                       endTime: Long)(implicit spark: SparkSession): MappingSummaryData = {
+                       endTime: Long,
+                       attemptedCount: Long,
+                       validRecordCount: Long)(implicit spark: SparkSession): MappingSummaryData = {
     import spark.implicits._
 
     // these three Encoders allow us to tell Spark/Catalyst how to encode our data in a DataSet.
     // val oreAggregationEncoder: ExpressionEncoder[Row] = RowEncoder(model.sparkSchema)
 
-    // val results: Dataset[Row] = mappingResults.map(oreAgg => oreAgg)(oreAggregationEncoder)
-
-    val messages = MessageProcessor.getAllMessages(results)(spark)
-    val warnings = MessageProcessor.getWarnings(messages)
-    val errors =   MessageProcessor.getErrors(messages)
+    // Transformation
+    val messages: DataFrame = MessageProcessor.getAllMessages(results)(spark)
+      .persist(StorageLevel.MEMORY_AND_DISK_SER)
+      .repartition(col("id"))
```

Comment:
Are you sure about calling repartition() after persist()? Seems like you'd be repartitioning over and over potentially...

Reply:
Agree, plus there are no discernible performance improvements.  Removed.